privilege/privileges/privileges.go

```diff
@@ -198,6 +198,14 @@ func (p *UserPrivileges) isValidHash(record *UserRecord) bool {
 		return false
 	}
 
+	if record.AuthPlugin == mysql.AuthSM3Password {
+		if len(pwd) == mysql.SM3PWDHashLen {
+			return true
+		}
+		logutil.BgLogger().Error("user password from system DB not like a sm3_password format", zap.String("user", record.User), zap.String("plugin", record.AuthPlugin), zap.Int("hash_length", len(pwd)))
```

Comment:
I think the message could be improved. However an almost similar message is used for the other authentication types.
```suggestion
		logutil.BgLogger().Error("the password from the mysql.user table does not match the definition of a tidb_sm3_password", zap.String("user", record.User), zap.String("plugin", record.AuthPlugin), zap.Int("hash_length", len(pwd)))
```

Maybe we could unify/standardize the message as well.